tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -328,4 +364,75 @@ def montecarlo_main_loop(
         virt_packet_last_interaction_type,
         virt_packet_last_line_interaction_in_id,
         virt_packet_last_line_interaction_out_id,
+        rpacket_collections,
+    )
+
+
+def create_tracked_rpacket_df(rpacket_collections, iteration):
+    for i in range(len(rpacket_collections)):
+        if i == 0:
+            seed = rpacket_collections[0].seed
+            index = rpacket_collections[0].index
+            status = rpacket_collections[0].status
+            r = rpacket_collections[0].r
+            nu = rpacket_collections[0].nu
+            mu = rpacket_collections[0].mu
+            energy = rpacket_collections[0].energy
+            shell_id = rpacket_collections[0].shell_id
+
+        seed = [*seed, *rpacket_collections[i].seed]
```

Comment:
Look into `itertools.chain`

Reply:
Thank you for this. This works as intended :)